opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java

```diff
@@ -99,12 +99,10 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
             }
         }
 
-        ActiveSpan activeSpan = tracer.activeSpan();
-        if (activeSpan != null) {
-            activeSpan.deactivate();
-        } else {
-            spanWrapper.finish();
+        if (tracer.activeSpan() != null) {
```

Comment:
The deactivate method sets back the parent span as the active span, which I think is not done here. Let me know what you think.

Reply:
Yes, this could cause problems if there is a filter registered before this. 

I have pushed a fix: capture to prevent finishing and then close